haptics/src/web.ts

```diff
@@ -0,0 +1,40 @@
+import { WebPlugin } from '@capacitor/core';
+import {
+  HapticsPlugin,
+  HapticsImpactOptions,
+  HapticsNotificationOptions,
+  VibrateOptions,
+} from './definitions';
+
+export class HapticsWeb extends WebPlugin implements HapticsPlugin {
+  constructor() {
+    super({ name: 'Haptics' });
+  }
+
+  impact(options: HapticsImpactOptions): void {
+    console.log('impact options', options);
+  }
+
+  notification(options: HapticsNotificationOptions): void {
+    console.log('notification options', options);
+  }
+
+  vibrate(options?: VibrateOptions): void {
+    if (navigator.vibrate) {
+      const duration = options?.duration || 300;
+      navigator.vibrate(duration);
+    }
```

Comment:
Is this usually how we handle feature detection? In the Accessibility plugin, we reject with an error: https://github.com/ionic-team/capacitor/blob/main/core/src/web/accessibility.ts#L21-L23

Reply:
At the moment none of the Haptics methods return anything, so we can't reject it. But we can change it to return promises as other plugins do and then reject when appropriate.

Reply:
We could also `throw` if the method is synchronous. 

We have two choices when features in a particular browser aren't supported:
1. Be silent about the missing feature and do nothing (perhaps log a warning). 
1. Throw an error (or reject) about the missing feature.

I think we should choose one of these and be consistent throughout all plugins.